manifests/install.pp

```diff
@@ -1,6 +1,13 @@
 # Katello Install
 class katello::install {
-  package { $katello::package_names:
-    ensure => installed,
+
+  if $katello::packages == undef {
```

Comment:
You can also solve this using [pick](https://github.com/puppetlabs/puppetlabs-stdlib#functions):
```puppet
$packages = pick($katello::packages, $katello::package_names)
package { $packages:
  ensure => installed,
}
```